frontend/src/Onboarding/TransferCoursesScreen.tsx

```diff
@@ -35,253 +30,159 @@ const TitleText = styled.div`
- * @param reqs
- */
-function flatten(reqs: Requirement[]): IRequiredCourse[][] {
-  return reqs.map(flattenOne).reduce((array, cur) => array.concat(cur), []);
-}
-
-function flattenOne(req: Requirement): IRequiredCourse[][] {
-  if (req.type === "COURSE") {
-    return [[req as IRequiredCourse]];
-  } else if (
-    req.type === "AND" &&
-    req.courses.filter(c => c.type === "COURSE").length
-  ) {
-    return [req.courses as IRequiredCourse[]];
-  } else if (req.type === "AND" || req.type === "OR") {
-    return flatten(req.courses);
-  } else {
-    return [];
-  }
-}
-
-interface TransferCoursesScreenProps {
+type TransferCoursesScreenProps = {
   major: Major;
   completedRequirements: IRequiredCourse[];
   setCompletedCourses: (completedCourses: ScheduleCourse[]) => void;
   setTransferCourses: (transferCourses: ScheduleCourse[]) => void;
-}
-
-type Props = TransferCoursesScreenProps & RouteComponentProps;
-
-interface State {
-  selectedRequirements: IRequiredCourse[];
-  completedNonTransfer: IRequiredCourse[];
-  modalVisible: boolean;
-  otherCourses: IRequiredCourse[][];
-}
-
-class TransferCoursesComponent extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      modalVisible: false,
-      selectedRequirements: [],
-      otherCourses: [],
-      completedNonTransfer: props.completedRequirements,
-    };
-  }
-
-  hideModal() {
-    this.setState({ modalVisible: false });
-  }
-
-  showModal() {
-    this.setState({ modalVisible: true });
-  }
-
-  // Adds the "other courses" to the state in the form of a IRequiredCourse[][] so that they can be
-  // processed by renderAllCourses
-  addOtherCourses(courses: ScheduleCourse[]) {
+} & RouteComponentProps;
+
+const coursesToString = (c: IRequiredCourse[]) =>
+  c.map(courseToString).join(",");
+const mappableFetchCourse = (c: IRequiredCourse) =>
+  fetchCourse(c.subject, String(c.classId));
+const fetchAndFilterCourses = async (cs: IRequiredCourse[]) => {
+  const courses = await Promise.all(cs.map(mappableFetchCourse));
+  // filter will get rid of null values, so cast is safe
+  return courses.filter(c => c !== null) as ScheduleCourse[];
+};
+
+const courseEq = (c1: IRequiredCourse, c2: IRequiredCourse) =>
+  c1.classId === c2.classId && c1.subject === c2.subject;
+
+const TransferCoursesScreenComponent: React.FC<TransferCoursesScreenProps> = () => {
+  const major = useSelector((state: AppState) => getUserMajorFromState(state)!);
+  const completedRequirements = useSelector((state: AppState) =>
+    getCompletedRequirementsFromState(state)
+  );
+  const dispatch = useDispatch();
+  const setCompletedCourses = (completedCourses: ScheduleCourse[]) =>
+    dispatch(setCompletedCoursesAction(completedCourses));
+  const setTransferCourses = (transferCourses: ScheduleCourse[]) =>
+    dispatch(setTransferCoursesAction(transferCourses));
```

Comment:
first time using redux 🤪 so lmk if this looks ok or not

Reply:
ooh I like `useDispatch` and `useSelector`! Curious are there any benefits to using these as opposed to `connect` or is this just the more functional/hooks way of doing things?

Reply:
yes, this is just the functional way of doing things. and there aren't "benefits" per se, but the API is slightly different/more flexible, see https://react-redux.js.org/api/hooks#useselector